### PR TITLE
adding custom_fields support for the API

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Adding support for CustomFields:
1) added custom_fields_mapping to params - this is a dict where the keys are the **current** field names and the values are the **new** field names.

2) added function create_custom_fields which gets the rawjson dict and creates the CustomFields dict in accordance with the custom_field_mapping.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 1]
```diff
@@ -214,6 +217,18 @@ def build_iterator(self, **kwargs):
...
+        created_custom_fields = {}
+        for attribute in attributes.keys():
+            if attribute in self.custom_fields_mapping.keys():
+                created_custom_fields[self.custom_fields_mapping[attribute]] = attributes[attribute]
+
+        if len(created_custom_fields.keys()) > 0:
+            return None
+
+        else:
+            return created_custom_fields
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py]: why do we need the elif? shouldn't it just be `return created_custom_fields` ?

**Comment by liorblob** [comment]
@roysagi Can you add a unit test for this?

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py (hunk 1 above)
Comment (roysagi) [reply Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py]: done

**Comment by roysagi** [comment]
> @roysagi Can you add a unit test for this?

done

**Comment by roysagi** [comment]
no documentation required so adding docs approved label

**Comment by ronykoz** [comment]
Force merging due to unrelated problem

**Pull request merged**